File: company-implementation/src/main/java/telran/employees/Manager.java
package telran.employees;

import org.json.JSONObject;

public class Manager extends Employee{
    private float factor;
    public Manager(){}
    public Manager(long id, int basicSalary, String department, float factor){
        super(id, basicSalary,department);
        this.factor = factor;
    }
    @Override
    public int computeSalary() {
        return (int)(super.computeSalary() * factor);
    }
    public float getFactor() {
        return factor;
    }
    
    @Override
    protected void fillJSON(JSONObject jsonObj) {
        super.fillJSON(jsonObj);
        jsonObj.put("factor", factor);

    }
    @Override
    protected void setObject(JSONObject jsonObj) {
        super.setObject(jsonObj);
        factor = jsonObj.getFloat("factor");
     }
}

File: Node-Project/src/Manager.mjs
import Employee from "./Employee.mjs";
export default class Manager extends Employee {
    #factor;
    constructor(id, department, basicSalary, factor=0) {
        super(id, department, basicSalary);
        this.#factor = factor;
    }
    getFactor() {
        return this.#factor;
    }
    computeSalary() {
        return super.computeSalary() * this.#factor;
    }

}